go/adbc/driver/snowflake/record_reader.go

```diff
@@ -212,13 +215,7 @@ func getTransformer(sc *arrow.Schema, ld gosnowflake.ArrowStreamLoader, useHighP
 							continue
 						}
 
-						q := int64(t) / int64(math.Pow10(int(srcMeta.Scale)))
-						r := int64(t) % int64(math.Pow10(int(srcMeta.Scale)))
-						v, err := arrow.TimestampFromTime(time.Unix(q, r), dt.Unit)
-						if err != nil {
-							return nil, err
-						}
-						tb.Append(v)
+						tb.Append(arrow.Timestamp(t))
```

Comment:
I'm confused, did Snowflake change how they return timestamp data?  In addition, this change to use `tb.Append(arrow.Timestamp(t))` doesn't take into account the timestamp unit for the input data type.

Reply:
It's entirely possible that they did, the behavior seems to vary between different types (such as TIMESTAMP_TZ vs TIMESTAMP_LTZ). When I switched the type to TIMESTAMP_LTZ, all results were returned in this format rather than the struct representation.

In this particular case, I observed that the returned Int64 value corresponds to the scalar value of the timestamp in whatever unit is specified by the type. If the scale is 3, then the Int64 is already meant to denote milliseconds, and so on. Since the values are already in the unit specified by scale, we can just add the existing value to the array. The test [TestSqlIngestTimestampTypes](https://github.com/apache/arrow-adbc/pull/1456/files#diff-62fd1aaf4052e89445863d9d98aad5b6ff9b92e737f95e2700e516ecf965fddeR612) confirms the roundtrip behavior (currently skipped but it does pass when the upstream changes are pulled in).

Reply:
> Since the values are already in the unit specified by scale, we can just add the existing value to the array

This is assuming the unit for the arrow column matches the scale that snowflake is returning which i'm not sure is a valid assumption. We should ensure we scale the value properly to the desired arrow unit. (which is what the previous code did). Are we sure we can always assume the unit for the Arrow column will match the scale that Snowflake returned?

Reply:
I think we might have to make this assumption one way or another. In the existing approach we rely on `srcMeta.Scale` being correct, which comes from Snowflake. If that doesn't match the actual scale of the `int64` value we get then the calculations to get `q` and `r` (`sec` and `nsec`) will be wrong as well. Given that assumption, the additional computation doesn't end up being necessary.